internal/gorson/io/io.go

```diff
@@ -135,9 +152,15 @@ func WriteToParameterStore(parameters map[string]string, path util.ParameterStor
 	// and we error out
 	select {
 	case <-done:
-		return
+		// if any results came back with errors, return the first error
+		for _, result := range results {
+			if result.Error != nil {
+				return result.Error
+			}
+		}
+		return nil
```

Comment:
this is probably not the perfect way to handle errors - we know we have an error earlier if one of the goroutines returns an error, but we're still waiting to get all results (or timeout after a minute) before we let the caller know.

I don't think it's a big deal on this scale, but there's probably a better pattern to find.

Reply:
Is this a potential solution? https://github.com/pbs/gorson/pull/37